packages/inventory-compliance/src/SystemRulesTable.js

```diff
@@ -473,25 +501,54 @@ class SystemRulesTable extends React.Component {
         this.uncompressRows(this.compressRows(rows))
     )
 
-    updateFilter = (hidePassed, severity, policy) => {
+    getSelectedRows = (rows) => {
+        const { selectedRefIds } = this.state;
+        let selectedRows = [];
+        rows.forEach((row, i) => {
+            if (this.isParent(row) && selectedRefIds.includes(this.rowRefId(row))) {
+                selectedRows.push(row);
+                if (!this.isParent(rows[i + 1])) {
+                    selectedRows.push({
+                        parent: selectedRows.length - 1,
+                        cells: rows[i + 1].cells
+                    });
+                }
+            }
+        });
+        return selectedRows;
+    }
+
+    markSelectedRows = (rows) => {
+        const { selectedRefIds } = this.state;
+        return rows.map(row => (
+            {
+                ...row,
+                ...this.isParent(row) && selectedRefIds.includes(this.rowRefId(row)) ? { selected: true } : {}
+            }
+        ));
+    }
+
+    updateFilter = (hidePassed, severity, policy, selectedFilter) => {
         const { columnIndices, originalRows, profiles, refIds, itemsPerPage, searchTerm } = this.state;
+
         let passedRows;
         if (hidePassed) {
             passedRows = this.filterBy(!hidePassed, originalRows, columnIndices.COMPLIANT_COLUMN);
         } else {
             passedRows = originalRows;
         }
 
+        const selectedRows = this.getSelectedRows(originalRows);
         const severityRows = this.filterBy(severity.join('|'), originalRows, columnIndices.SEVERITY_COLUMN);
         const policyRows = this.filterBy(policy, originalRows, columnIndices.POLICY_COLUMN);
         const searchRows = this.searchBy(searchTerm, originalRows);
-        const filteredRows = this.filteredRows(passedRows, severityRows, policyRows, searchRows,
-            hidePassed, severity, policy, searchTerm);
+        const filteredRows = this.filteredRows(passedRows, severityRows, policyRows, searchRows, selectedRows,
+            hidePassed, severity, policy, searchTerm, selectedFilter);
 
         return this.setState({
-            currentRows: this.currentRows(1, itemsPerPage, { rows: filteredRows, profiles, refIds }),
+            currentRows: this.markSelectedRows(this.currentRows(1, itemsPerPage, { rows: filteredRows, profiles, refIds })),
```

Comment:
Before displaying selected rows, let's mark them as selected based on the `selectedRefIds` array